packages/datadog-instrumentations/src/next.js

```diff
@@ -212,3 +214,36 @@ addHook({
 
   return nextServer
 })
+
+addHook({
+  name: 'next',
+  versions: ['>=13'],
+  file: 'dist/server/web/spec-extension/request.js'
+}, request => {
+  const nextUrlDescriptor = Object.getOwnPropertyDescriptor(request.NextRequest.prototype, 'nextUrl')
+  shimmer.wrap(nextUrlDescriptor, 'get', function (originalGet) {
+    return function wrappedGet () {
+      const nextUrl = originalGet.apply(this, arguments)
+      if (queryParsedChannel.hasSubscribers) {
+        queryParsedChannel.publish({
+          query: Object.fromEntries(nextUrl.searchParams)
```

Comment:
For query strings that informs arrays, like `?param[]=foo&param[]=bar`, it does not work, since the result object will only contains the latest `{ 'param[]': 'bar' }`

Reply:
oh, good catch!

Reply:
changed and added tests